bin/generate

```diff
@@ -39,11 +43,11 @@ func curryDefinitionGenerator(arguments: Int) -> String {
   let lowerFunctionArguments = inputParameters.map { "\($0.lowercased())" } // ["a", "b", "c"]
   let returnType = genericParameters.last! // "D"
 
-  let functionArguments = "(\(commaConcat(inputParameters)))" // "(A, B, C)"
+  let functionArguments = extraParentheses("(\(commaConcat(inputParameters)))", when: inputParameters.count > 1) // "(A, B, C)"
```

Comment:
yep, you get something like `Cannot convert value of type '(String) -> Service1' to expected argument type '((_)) -> _'`. There can't be extra parentheses for a single parameter.